### ISSUE TITLE
jupyter-notebook needs json >= 1.5.0

### ISSUE THREAD

**Issue by lostbard** [gnu-octave/pkg-jupyter-notebook#8]
Octave 7 included json functions already so shouldnt need the json package as a dependancy

The json package also wont install due to the functions already available in octave.

**Comment by siko1056** [comment]
This is true.  I think Octave 7 is already wide spread enough to safely assume `jsonencode` and `jsondecode` are present in Octave.  If not, a small help text could be provided.

@Abdallah-Elshamy are you interested in working on this?

**Comment by Abdallah-Elshamy** [comment]
I am interested. A simple solution for this would be simply to change the requirements to:
```
depends: octave (>= 7.1.0)
```
However this will not work for older versions' users that use the `json` package. 

Another solution will be to remove `depends` and make the check in another place. I am not sure where is the best place to do so: 

* One simple place is at the constructor of the class by checking `exist("jsonencode")` but this will not prevent the package from being installed without the needed functions. 
* Another possible place is a `package/pre_install.m` script that does that. But, I am not sure if this kind of scripts is suitable for such checks as I did not use them before

**Comment by victoryred** [comment]
I downloaded and configured the octave-9.4.0 src for a linux build.
I just located these two file in ~/octave-9.4.0/.build/libinterp/corefcn;

libcorefcn_la-jsonencode.lo
libcorefcn_la-jsondecode.lo

So I suggest that someone updates the "https://github.com/gnu-octave/pkg-jupyter-notebook/archive/v1.1.0.tar.gz"
tarball to include these files. If not why not?

**Comment by siko1056** [comment]
Hello @victoryred ,
This package is already included into Octave core.  The only function of this package `jupyter_notebook` should be included in Octave version 7 and newer: https://octave.org/NEWS-7.html#general-improvements

I will add this note to the README-file to avoid further confusion about this matter.

**Comment by siko1056** [comment]
Technically and by documentation covered here: https://github.com/gnu-octave/pkg-jupyter-notebook/releases/tag/v1.3.0

**Comment by victoryred** [comment]
Hello Kai,So good to hear from you.The Depends on JSON(>1.5) stops the build process. I don’t have JSON on my machine and it’s not available in apt, pip etc. Here’s what I would like to do in the end. I want to add octave as a back end for LabPlot2. I can’t find how to do it. Labplot2 says to make a notebook for Octave. I can’t find anything else on how to do that.

**Comment by siko1056** [comment]
@victoryred Does your operating system distribution contain a RapidJSON package like https://packages.ubuntu.com/noble/rapidjson-dev

When you compiled Octave from source, what was the output about looking up the RapidJSON library?

**Comment by victoryred** [comment]
Unfortunately I am in the hospital right now. I didn’t load manually but I can go through my package managers when I get home. Normally would someone use apt or pip3 to get it.Sent from the Luminescent EtherOn Mar 25, 2025, at 2:30 PM, Kai Torben Ohlhus ***@***.***> wrote:﻿
Reopened #8.
